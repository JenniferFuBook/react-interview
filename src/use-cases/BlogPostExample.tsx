import BlogPost from '../components/blog-post/BlogPost';
import { samplePost } from './data';

const BlogPostExample = () => <BlogPost post={samplePost} />;

export default BlogPostExample;
