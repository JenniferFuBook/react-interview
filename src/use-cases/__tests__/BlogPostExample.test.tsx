import { render, screen } from '@testing-library/react';
import BlogPostExample from '../BlogPostExample';

describe('BlogPostExample', () => {
  it("matches snapshot", () => { // This is a snapshot test
    const { container } = render(<BlogPostExample />);
    expect(container).toMatchSnapshot();
  });

  it('renders the blog post title', () => {
    // This test checks if the title of the blog post is rendered correctly
    render(<BlogPostExample />);
    const titleElement = screen.getByText(/My Favorite Article/i);
    expect(titleElement).toBeInTheDocument();
  });

  it('renders the blog post content', () => { 
    // This test checks if the content of the blog post is rendered correctly
    render(<BlogPostExample />);
    const contentElement = screen.getByText(/React is a JavaScript library/i);
    expect(contentElement).toBeInTheDocument();
  });
});
