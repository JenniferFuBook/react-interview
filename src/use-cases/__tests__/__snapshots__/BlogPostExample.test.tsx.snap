// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPostExample matches snapshot 1`] = `
<div>
  <main>
    <article>
      <header>
        <h1>
          My Favorite Article
        </h1>
        <h2>
          A demonstration of semantic HTML elements
        </h2>
      </header>
      <section>
        <h3>
          Introduction
        </h3>
        <div>
          <h4>
            What is React?
          </h4>
          <p>
            React is a JavaScript library for building user interfaces. React apps are made out of components. A component is a piece of the UI (user interface) that has its own logic and appearance. A component can be as small as a button, or as large as an entire page.
          </p>
        </div>
        <div>
          <h4>
            Why use React?
          </h4>
          <p>
            React makes it painless to create interactive UIs. Design simple views for each state in your application, and React will efficiently update and render just the right components when your data changes. Declarative views make your code more predictable and easier to debug.
          </p>
        </div>
      </section>
    </article>
  </main>
  <footer>
    <p>
      Â© 2025 My Website. All rights reserved.
    </p>
  </footer>
</div>
`;
